### PR TITLE
change: flesh out SageMaker training integ tests

### PR DESCRIPTION
**Pull request by laurenyu**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a37a70b8-8ff0-4f66-ac83-cd6b7c6b6b63%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: test/integration/sagemaker/test_training.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from test.integration import RESOURCE_PATH
+from timeout import timeout
+
+DATA_PATH = os.path.join(RESOURCE_PATH, 'mnist')
+SCRIPT_PATH = os.path.join(DATA_PATH, 'mnist.py')
+
+
+@pytest.mark.deployment_test
+def test_single_machine(sagemaker_session, ecr_image, instance_type):
+    _fit_estimator(1, instance_type, sagemaker_session, ecr_image)
```
Comment (icywang86rui) [inline test/integration/sagemaker/test_training.py]: Optional - we can make the instance count a comma delimited parameter, something like this:

```
--instance-count 1,2
```

Comment (laurenyu) [reply test/integration/sagemaker/test_training.py]: done. I'll probably do something similar for Python version and processor in a later PR

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=275706ae-c1ac-40ff-8a00-1759387adf04%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**